web/docs/understand/data-model/type-system.md

```diff
@@ -6,35 +6,298 @@ sidebar_position: 0
+
+```yaml
+enum:
+  - on
+  - off
+```
+
+### List Type
+
+Use the `list` key to define list with a corresponding element type on the
+right-hand side:
+
+```yaml
+list: url
+```
+
+Lists nest naturally:
+
+```yaml
+list:
+  list:
+    enum:
+      - on
+      - off
+```
+
+### Record Type
+
+Use the `record` key to define a list of fields:
+
+```yaml
+record:
+  - src_ip: ip
+  - dst_ip: ip
+  - src_port: port # existing alias
+  - dst_port: port # existing alias
+  - proto: string
+```
+
+Records nest naturally:
+
+```yaml
+record:
+  - src:
+      record:
+        - ip: ip
+        - port: uint64
+  - dst:
+      record:
+        - ip: ip
+        - port: uint64
+  - proto: string
+```
+
+We can also factor nested records into dedicated types:
+
+```yaml
+endpoint:
+  record:
+    - ip: ip
+    - port: uint64
+```
+
+Thereafter we can reference the `endpoint` type by its name:
+
+```yaml
+record:
+  - src: endpoint
+  - dst: endpoint
+```
+
+Records of lists:
+
+```yaml
+record:
+  - values:
+      list: string
+```
+
+Lists of records:
+
+```yaml
+list:
+  record:
+    - question: string
+    - correct: bool
+```
+
+#### Record Algebra
+
+In addition to nesting records, a frequent use case is embedding fields of
+one record into another. This occurs often with families of event types that
+share a large portion of fields. Algebraically, we want to "add" the two records
+together to combine their field.
+
+For example, consider this `common` record with a single field `shared`:
+
+```yaml
+# The base record.
+common:
+  record:
+    - shared:
+        type: bool
+```
+
+To embed `common` in another record, reference it using the `base` key. This
+requires moving the fields one layer down under the `fields` key:
+
+```yaml
+record:
+  base:
+    - common
+  fields:
+    - unique: string
+```
+
+The effect is the same specifying a single record with two fields:
+
+```yaml
+record:
+  - shared: bool
+  - unique: string
+```
+
+There are three special keys under the `record` key that control what to do when
+field name clashes occur between the two records:
+
+1. `base`: raise an error
+2. `implant`: prefer the base record
+3. `extend`: prefer the current record
+
+## Comparison to Arrow
+
+All VAST types have a lossless mapping to Arrow types, however, not all Arrow
+types have a VAST equivalent. As a result, it is currently not yet possible to
+import arbitrary Arrow data. In the future, we plan to extend our support for
+Arrow-native types and also offer conversion options for seamless data handover.
+
+VAST has a few domain-specific types that map to Arrow [extension
+types][extension-types]. These are currently:
+
+- `enum`
+- `ip`
+- `subnet`
 
 [extension-types]: https://arrow.apache.org/docs/format/Columnar.html#extension-types
 
+VAST treats type meta data differently from Arrow. In VAST, the type is the
+component that contains metadata. In Arrow, it's the record field. This makes
```

Comment:
It also makes it so that in VAST we don't need a special "schema" structure, because it's just a named record type. For Arrow, a schema is a struct type with metadata.

Reply:
Nice, I added that fact.